org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/annotations/typing/XAnnotationUtil.java

```diff
@@ -70,5 +73,21 @@ public XExpression findValue(XAnnotation annotation, JvmOperation jvmOperation)
 		}
 		return emptySet();
 	}
-	
+
+	/**
+	 * @since 2.15
+	 */
+	public List<JvmAnnotationReference> findAnnotations(Set<String> qualifiedNames, JvmAnnotationTarget annotationTarget) {
+		List<JvmAnnotationReference> result = new ArrayList<>();
+		
+		if(annotationTarget!=null) {
+			for(JvmAnnotationReference annotation : annotationTarget.getAnnotations()) {
+				String annotationQualifiedName = annotation.getAnnotation().getQualifiedName();
```

Comment:
I'm a little bit in favor of using getIdentifier here instead of getQualifiedName, since getIdentifier is cached information whereas the qualified name is usually recomputed.